### PR TITLE
Gallery entry for 2020-Jan calendar

### PR DESCRIPTION
**Pull request by smithWEBtek**
# Why is this change necessary?
Gallery entry for 2020-Jan calendar

# How does it address the issue?
Gallery entry for 2020-Jan calendar

* Add controller action
* Add route
* Add thumbnail image of NAICS Bar Chart Race
* Refactor Javascript errors to get chart to load
* Add CSV files to /public/data folder for loading

# What side effects does it have?
I have not successfully dealt with webpack loader, the files are in /public/data, which is not ideal

### CODE REVIEW

**Comment by smithWEBtek** [comment]
@mzagaja this is a start, the route, controller, react component data and csv files are working.  Still have to style it.

File: app/javascript/packs/naics-bar-chart-race.js [hunk 1]
```diff
@@ -0,0 +1,191 @@
+
+window.addEventListener('DOMContentLoaded', () => {
+  Promise.all([
+    d3.csv('/data/three-digit-sub-sector-01-17.csv', (d) => ({
```
Comment (mzagaja) [inline app/javascript/packs/naics-bar-chart-race.js]: Should be able to drop the source file in `/app/assets/data/file.csv` and access it via `/assets/file.csv`. For example try http://127.0.0.1:3000/assets/ma-munis.json when running the server locally. For this to work in production we may need to [disable asset fingerprinting](https://edgeguides.rubyonrails.org/asset_pipeline.html#what-is-fingerprinting-and-why-should-i-care-questionmark) as well. The other option would be to find a csv loader for webpack, but looking further into this it seems like it does some things before we would hit D3 and that is not a rabbit hole I want to jump down anymore.

File: app/javascript/packs/naics-bar-chart-race.js [hunk 2]
```diff
@@ -0,0 +1,191 @@
...
+      measurementDate: +d.date,
+      categorys: d.category,
+      municipality: d.name,
+      employees: d.employees
```
Comment (mzagaja) [inline app/javascript/packs/naics-bar-chart-race.js]: This should be updated to `+d.employees` to coerce it into the correct integer type to fix the issue with the type conversion that was causing the incorrect rankings.

File: app/views/calendars/2020-01.html.erb [hunk 3]
```diff
@@ -0,0 +1,7 @@
+    <h1>NAICS Bar Chart Race</h2>
+    <div class="naics-bar-chart-race"></div>
+
+    <script src="https://cdnjs.cloudflare.com/ajax/libs/d3/5.14.2/d3.min.js"></script>
```
Comment (mzagaja) [inline app/views/calendars/2020-01.html.erb]: I think you should be able to add [d3 with yarn](https://yarnpkg.com/en/package/d3) in `naics-bar-chart-race.js`. Try it:

```
yarn add d3
```

Then add the line 
```
import * as d3 from "d3";
```
to `naics-bar-chart-race.js`.

File: app/controllers/resources_controller.rb [hunk 4]
```diff
@@ -58,6 +58,10 @@ def destroy
     redirect_to resources_path
   end
 
+  def calendars
+    render "/calendars/#{params[:id]}"
```
[the remark this answers is not in the document]
Comment (mzagaja) [reply app/controllers/resources_controller.rb]: So after morning standup I think that what we're going to want to do here is use the resources show view for these and point the calendar to the resource show.

**Pull request closed without merging**